hub/core/meta/encode/base_encoder.py

```diff
@@ -11,6 +11,26 @@
 
 
 class Encoder(ABC):
+    last_row = 0
+
+    def is_index_in_last_row(self, arr, index) -> bool:
```

Comment:
the logic to skip binary search in chunk engine got lost somehow when tiling PR was merged. I've moved it to encoder now so all encoders can use it.